vector.go

```diff
@@ -28,6 +28,9 @@ type vector struct {
 	// The child vectors of nested data types.
 	childVectors []vector
 
+	// FIXME: this is a workaround until the C API exposes SQLNULL.
+	isSQLNull bool
```

Comment:
It would be nice to have some more documentation on what this actually does.

Does this also impact table functions? if so it would be nice to know so I can use it too

Reply:
The input parameter can have no type, for example: `SQLNULL`, `LIST(SQLNULL)`, `STRUCT(VARCHAR: SQLNULL)`.
```sql
SELECT my_func(NULL, [NULL], {'hello': NULL});
```

So, `isSQLNull` is set when inferring the type from a DuckDB data chunk. Currently, we see `INVALID` as the type because the C API does not (yet) expose `SQLNULL`: https://github.com/duckdb/duckdb/pull/13999.

This is only relevant for scenarios where untyped input is possible. In the value getter and setter functions, we treat `SQLNULL` as `nil`.

Reply:
I've elaborated on the `FIXME`.

Reply:
Ah so it is a type indicating the entire column is `NULL`? That sounds nice to know indeed.

Reply:
Exactly. 👍